src/async.zsh

```diff
@@ -0,0 +1,53 @@
+
+#--------------------------------------------------------------------#
+# Async                                                              #
+#--------------------------------------------------------------------#
+
+_zsh_autosuggest_async_fetch_suggestion() {
+	local strategy_function="_zsh_autosuggest_strategy_$ZSH_AUTOSUGGEST_STRATEGY"
+	local prefix="$(_zsh_autosuggest_escape_command "$1")"
+
+	# Send the suggestion command to the pty to fetch a suggestion
+	zpty -w -n $ZSH_AUTOSUGGEST_PTY_NAME "$strategy_function '$prefix'"$'\0'
+}
+
+_zsh_autosuggest_async_suggestion_worker() {
+	local last_pid
+
+	while read -d $'\0' cmd; do
+		# Kill last bg process
+		kill -KILL $last_pid &>/dev/null
+
+		# Run suggestion search in the background
+		print -n -- "$(eval "$cmd")"$'\0' &
+
+		# Save the bg process's id so we can kill later
+		last_pid=$!
+	done
+}
+
+_zsh_autosuggest_async_suggestion_ready() {
+	# while zpty -rt $ZSH_AUTOSUGGEST_PTY_NAME suggestion 2>/dev/null; do
+	while read -u $_ZSH_AUTOSUGGEST_PTY_FD -d $'\0' suggestion; do
```

Comment:
So I'm playing with `zpty -r` over here, and I'm seeing some non-printable characters showing up in my output. 

<img width="378" alt="screenshot 2017-01-24 17 51 09" src="https://cloud.githubusercontent.com/assets/79998/22273398/c157ca6c-e25d-11e6-9e25-8edbd4f5e92e.png">

<img width="382" alt="screenshot 2017-01-24 17 51 21" src="https://cloud.githubusercontent.com/assets/79998/22273399/c3a7bf16-e25d-11e6-9ef4-454babbc907e.png">

You ever seen this? I can probably remove them with a regex or something, but I don't quite understand where these chars are coming from.